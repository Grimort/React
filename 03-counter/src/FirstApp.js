// import { Fragment } from "react";
// SAME AS BELOW
// const FirstApp = () => {
//   return (
//     <Fragment>
//       <h1> Hello World !!!</h1>
//     </Fragment>
//   );
// };

import PropTypes from "prop-types";

export const FirstApp = ({ hi, subtitle }) => {
  const item = "Hello world";
  return (
    <>
      <h1>{item} !!!</h1>
      <h1>{hi} </h1>
      <p>{subtitle}</p>

      {/* <pre> {JSON.stringify(item)} !!!</pre> */}
      {/* <p> - My First APP -</p> */}
    </>
  );
};

FirstApp.propTypes = {
  hi: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  subtitle: "Default Subtitle",
};

export default FirstApp;
