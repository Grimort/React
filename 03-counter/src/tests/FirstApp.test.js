import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import FirstApp from "../FirstApp";

describe("Testing FirstApp", () => {
  // before enzyne
  //   test('should to show a message "Hello world"', () => {
  //     const hi = "Hello world";
  //     const { getByText } = render(<FirstApp hi={hi} />);
  //     expect(getByText(hi)).toBeInTheDocument();
  //   });

  test("should to show FirstApp correctly", () => {
    const hi = "Hello world";
    const wrapper = shallow(<FirstApp hi={hi} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should to show FirstApp correctly with default subtitle", () => {
    const hi = "Hello world";
    const subtitle = "Default Subtitle";
    const wrapper = shallow(<FirstApp hi={hi} subtitle={subtitle} />);
    const paragrah = wrapper.find("p").text();
    console.log(paragrah);
    expect(paragrah).toBe(subtitle);
    expect(wrapper).toMatchSnapshot();
  });
});
