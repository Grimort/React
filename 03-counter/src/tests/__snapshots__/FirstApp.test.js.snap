// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FirstApp should to show FirstApp correctly 1`] = `
<Fragment>
  <h1>
    Hello world
     !!!
  </h1>
  <h1>
    Hello world
     
  </h1>
  <p>
    Default Subtitle
  </p>
</Fragment>
`;

exports[`Testing FirstApp should to show FirstApp correctly with default subtitle 1`] = `
<Fragment>
  <h1>
    Hello world
     !!!
  </h1>
  <h1>
    Hello world
     
  </h1>
  <p>
    Default Subtitle
  </p>
</Fragment>
`;
